Runtime/Shaders/MapsBaker.cs

```diff
@@ -0,0 +1,235 @@
+
+            Texture2DDisposable rgbTexture = null;
+            string channelName = null;
+
+            bool texturesExist = false;
+
+            if (smoothnessChannel == 0) {
+                // Bake to metallic alpha
+                channelName = MeshSyncConstants._MetallicGlossMap;
+                texturesExist |=
+                    FindTexture(MeshSyncConstants._MetallicGlossMap, textureHolders, materialProperties,
+                        MeshSyncConstants._Metallic, 0, out rgbTexture);
+            }
+            else if (smoothnessChannel == 1) {
+                // Bake to albedo alpha
+                channelName = MeshSyncConstants._BaseMap;
+                texturesExist |=
+                    FindTexture(MeshSyncConstants._MainTex, textureHolders, materialProperties, MeshSyncConstants._Color, 0,
+                        out rgbTexture);
+            }
+            else {
+                Debug.LogError(
+                    $"Unknown smoothness channel, cannot bake to option: {smoothnessChannel} set in the smoothness texture channel.");
+                return;
+            }
+
+            // Bake smoothness to 1 if there is no map and use the slider to scale it:
+            texturesExist |=
+                FindTexture(MeshSyncConstants._GlossMap, textureHolders, materialProperties, string.Empty, 1,
+                    out var glossTexture);
+
+            // If there are no textures, don't bake anything, slider values can control everything:
+            if (!texturesExist ||
+                glossTexture.Texture == null ||
+                rgbTexture.Texture == null) {
+                destMat.SetTexture(channelName, null);
+
+                if (channelName == MeshSyncConstants._MetallicGlossMap) {
+                    destMat.DisableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                    destMat.DisableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+                }
+
+                return;
+            }
+
+            var shader = LoadShader(SHADER_NAME_SMOOTHNESS_INTO_ALPHA);
+            if (shader == null) {
+                return;
+            }
+
+            shader.SetTexture(SHADER_CONST_SMOOTHNESS, glossTexture.Texture);
+            shader.SetTexture(SHADER_CONST_RGB, rgbTexture.Texture);
+
+            var texture = shader.RenderToTexture(destMat.GetTexture(channelName));
+
+            if (channelName == MeshSyncConstants._MetallicGlossMap) {
+                destMat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                destMat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+            }
+            else if (smoothnessChannel == 0) {
+                destMat.DisableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                destMat.DisableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+            }
+
+            destMat.SetTexture(channelName, texture);
+
+            glossTexture.Dispose();
+            rgbTexture.Dispose();
```

Comment:
Do we need a null check here? Could the never get set?

Reply:
This is a Texture2DDisposable which is never null and internally releases the texture if it was set to do so.